Release 2.0

crhallberg opened the pull request:
I have a new version of the website and the generation tools coming out very soon.

- [x] Move examples to individual folders
- [x] Rename main files
- [x] Hide deck selection on mobile
    - [x] Try offscreen panel
    - ~~Base set, All Official, or EVERYTHING choices~~
    - ~~Add "more options on desktop" message~~

After merge:
- [x] Hacktoberfest label
- [x] Add issue to look into WebSQL

## Improvements

### Digesting all cards from the open spreadsheet

I've used [the massive open spreadsheet](https://docs.google.com/spreadsheets/d/1lsy7lIwBe-DWOi2PALZPf5DgXHx9MEvKfRw1GaWQkzg/edit#gid=1013305605) as a primary source for a long time and now I'm doing it automatically. **This will largely take the place of ingesting from the src folder documents**.

The digestion from src will remain in the repository to make it easy to create sets with custom decks.

### [Brand New Website](https://crhallberg.com/cah/v2/)

Modern look. More accessible. Faster deck downloading with local JS.

### [JS library](https://github.com/crhallberg/json-against-humanity/blob/v2/web/CAHDeck.js) to digest compact.md.json

```js
let deck = CAHDeck.fromCompact("./compact.md.json");
deck.listPacks() // Get metadata: [{ name: "CAH Base Pack", official: true, counts: { white: ...
deck.getPack(0); // Get one pack by index (based on listPacks)
deck.getPacks([0, 4]); // Combine packs by index (based on listPacks)
```

### Community Examples

Open source examples of using the cards data in various settings and systems. Starting with one:

#### [Canvas Deck Sampler](https://github.com/crhallberg/json-against-humanity/tree/v2/examples)
Demonstration of ingesting the compact.json file with the basic Javascript library and displaying cards on a canvas element.

## Breaking Changes

### Change to `compact.md.json`

In `cah-all-compact.json`, white cards will now just be a string instead of a JSON object.
```diff
{
  "cards": {
    "white": [
-     { "text": "A sexy saddle." },
+     "A sexy saddle.",
    ]
  }
}
```

### Removing `.html.txt` sources

They are currently not being used and this just creates more work. Will automate markdown to HTML for easier contribution if these files are demanded.

## Additional PRs after merge

- [x] Icons (separate pull request)
    - [ ] Make folder for icon set jsons
    - [ ] Make FontAwesome icon
    - [ ] Adjust script to allow compilation with an icon set (default to none)
        - [ ] Document CLI
        - [ ] Document icon locations
    - [ ] Make issue to source/generate SVGs of deck icons

- [ ] Custom deck inclusion
    - [ ] Create an example
    - [ ] Include decks from v1 not currently in the data set

Commit:
v2 init. Clear out miscellaneous and extraneous.

Commit:
Make compatible with python3.

Commit:
Create basic JS interface.

Commit:
Remove debug.

Commit:
Update README. Begin to change terminology to 'deck' for whole sha-bang and 'pack' for each individual set.

Commit:
Update LICENSE and FAQ.

Commit:
Add nicer formatted LICENSE.

Source: https://creativecommons.org/licenses/by-nc-sa/4.0/legalcode.txt

Commit:
Spruce up the place.

Commit:
Rename deck functions to pack.

Commit:
Load decks into buttons. Spruce up the place.

Commit:
Design checkout footer.

Commit:
Fix total updates.

Commit:
Add deck preview and canvas example.

Commit:
Fix bulk button icons.

Commit:
Update sources information.

Commit:
Standardize more JS to 'packs'.

Commit:
Add prefers-color-scheme-dark.

Commit:
Adjust text wrap the slightest bit.

Commit:
Remove card outlines from example.

Commit:
Example improvements and cleanup.

Commit:
Add title to example.

Commit:
Fix formatting.

Commit:
Activate toggles.

Commit:
Add download functionality.

Commit:
Add Google Sheets card parser.

Commit:
Move index.html to web folder.

Commit:
Normalize card characters with clean-text-utils.

Commit:
Remove debug.

Commit:
Expand nanoid for extra collision prevention.

Commit:
Export pack metadata as array, not map.

Commit:
Fix toggle/select buttons.

Commit:
Update meta.

Commit:
Properly employ Inter Variable.

Commit:
Minor style changes.

Commit:
Normalize pack names a bit.

Commit:
So... how many cards now?

(22,067 white, 6,235 black)

Commit:
Remove links to CAH and remove plea to give them money. Black Lives Matter. Improve lawyer speak.

Commit:
Normalize deck names on import, write to main compact.md.json.

Commit:
Start emoji icon selection.

Commit:
Add template credentials json for devs.

Commit:
Adjust example to work with new compact and without ids/abbrs.

Commit:
Update web to work with main compact and without ids/abbrs.

Commit:
Add examples to README.

Commit:
Lock line number links to a certain commit. This may be dumb.

robnester commented:
Depending on your workflow, `pandoc` can autoconvert md to html. This could be put in place as part of your commit or release pipeline.

crhallberg commented:
Going to drop this at the start of Hacktoberfest!

robnester commented on web/index.html:
```diff
@@ -54,26 +54,8 @@ <h2>FAQ</h2>
-
-              <p>This project meets the Share-Alike requirement with an updated Creative Commons (CC) license, the <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC&nbsp;BY-NC-SA&nbsp;4.0&nbsp;license</a>, as recommended by CC and allowed by CC's <a href="https://creativecommons.org/share-your-work/licensing-considerations/compatible-licenses">similar license clause</a>. If you have ...
-            </dd>
+            <dd>Very nice of you, but you legally can't. More on that later. You should instead donate to <a href="https://secure.actblue.com/donate/bail_funds_george_floyd">local community bail funds</a>, the <a href="https://www.naacpldf.org/">NAACP Legal Defense Fund</a>, and <a href="https://www.raceforward.org/">Race Forward</a>.</dd>
```
:raised_hands: Beautiful

Commit:
Move examples to individual folder.

Commit:
Update README.

Commit:
Remove icons pending future PR.

Commit:
Rename JSON output file for clarity. Remove full.json.

Commit:
Update README.
- Rename cah-all-compact.json
- Full is now generated via CAHDeck.js
- Remove icons... again?
- Update Fine Print and documention

Commit:
Fix import paths in web and examples.

Commit:
Improve mobile styling of footer.

Commit:
Push renamed file for v2.

Commit:
Improve sites and examples on mobile. Fix paths.

Commit:
Fix buttons being bound to wrong packs.

Commit:
Match README to web (more to do).

Commit:
When hydrating, excluding icon if unset.

Commit:
Merge remote-tracking branch 'origin/latest' into v2
